bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoHandler.java

```diff
@@ -111,64 +102,99 @@ public WemoHandler(Thing thing, UpnpIOService upnpIOService, WemoHttpCall wemoHt
     public void initialize() {
         Configuration configuration = getConfig();
 
-        if (configuration.get("udn") != null) {
-            logger.debug("Initializing WemoHandler for UDN '{}'", configuration.get("udn"));
-            service.registerParticipant(this);
-            onSubscription();
-            onUpdate();
+        if (configuration.get(UDN) != null) {
+            logger.debug("Initializing WemoHandler for UDN '{}'", configuration.get(UDN));
+            UpnpIOService localservice = service;
+            if (localservice != null) {
+                localservice.registerParticipant(this);
+            }
+            pollingJob = scheduler.scheduleWithFixedDelay(this::poll, 0, DEFAULT_REFRESH_INTERVALL_SECONDS,
+                    TimeUnit.SECONDS);
             updateStatus(ThingStatus.ONLINE);
```

Comment:
Can you try to set status to UNKNOWN here? Two reasons:
1. It would be more correct. Will be flipped to ONLINE when first data is received and we know for sure it's online.
2. It might fix tests.

Reply:
No, UNKNOWN is incorrect, OFFLINE would be a more appropriate choice

Reply:
Please read https://www.openhab.org/docs/concepts/things.html#thing-status. Other bindings will usually change status to **UNKNOWN** in `initialize()` until having confirmation that thing is online, for example by receiving first payload. The reason I mention this now is because I've had to read through a lot of code to help with this PR, and now I noticed this. It's relevant now because the tests are awaiting the thing to change status to **ONLINE**. But this happens prematurely, thus host is not yet initialized at this point. This flow was changed in this PR, and it could be one of the reasons why the tests are failing.

Reply:
Done for WemoHandler, If it is fine, will change other handler classes later

Reply:
Now the test complain about UNKNOWN in WemoHandler ….

Reply:
Hmm, I think you need to figure out how to run these tests since this trial-and-error approach seems hard. Perhaps it could even be possible by just creating a separate branch and letting the CI job on GitHub run it? Not sure.

For first failing test to match new logic, it would need to first await `ThingStatus.UNKNOWN` as this was changed. After `handleCommand()` it could then await `ThingStatus.ONLINE`, since we expect it to go online after that. And it shouldn't assert on `getHost()`. And just a note, we should not expecting anything from the polling job:

```
        // The device is registered as UPnP Device after the initialization, this will ensure that the polling job will
        // not start
        addUpnpDevice(SERVICE_ID, SERVICE_NUMBER, MODEL_NAME);
```

Reply:
I think we are both thinking to much "in real hardware".
I have removed the asserts now and as I have also added getHost() to initialize, during testsimulation it should be set straight away....
Let's see...

Reply:
Thanks for pointing me into the right direction, it was clearly commented in the tests.
getHost() needs a registered device to deliver a valid IP.
So what we can try is to add getHost() at top of handleCommand etc….
I will give this a try…

Reply:
New observation/hint: In `poll()` you have added:

```
                if (host.isEmpty()) {
                    host = getHost();
                }
```

But in `handleCommand()` just:

```
        if (host.isEmpty()) {
            logger.error("Failed to send command '{}' for device '{}': IP address missing", command,
                    getThing().getUID());
```

Test calls `handleCommand()`, but I don't know if it provides properly configured `UpnpIOService` with descriptorURL. So maybe `handleCommand()` could also try to initialize host by `getHost()`. However, I would then suggest to add the `host.IsEmpty()` check to `getHost()` itself and simply call `getHost()` whenever host is needed.

Reply:
Our messages crossed - we got the same idea at the same time. Some maybe it's the correct one. :-)

Reply:
Heureka !
Thats it, test hass passed !
Thanks for your patience….

Reply:
Congrats, that was hard to find! :-) I'm reopening comment in order to reconsider setting status **UNKNOWN** before leaving `initialize()`. That is what the comment was originally about, and it was changed back to **ONLINE** while trying to fix tests. Since it was already ONLINE before this PR, for me it's okay to leave it, and it can be taken up again in another iteration. Just wanted you to reconsider and resolve the comment for the right reason (e.g. "out of scope" or "don't want to change") rather than "changed back because of problems with tests". :-)

Reply:
I'd rather leave it ONLINE, as before this PR. UNKNOWN seems wrong to me.